### DISCUSSION TITLE
Disable opentelemetry exporter in Quarkus 3.x

### DISCUSSION THREAD

**Discussion by fabrii** [quarkusio/quarkus#34653] [Q&A] [answered]
Hi. In Quarkus 2.x we had `quarkus.opentelemetry.tracer.exporter.otlp.enabled` to disable the export of the traces.

How is this handled in 3.x? Do we have to set `quarkus.otel.traces.exporter` to none, or let `quarkus.otel.exporter.otlp.endpoint` unset?

Thanks

[a comment by an automated account is not shown]

**Reply by radcortez** [reply, marked as answer]
For compatibility you can use `quarkus.otel.exporter.otlp.enabled`. This is a build time property.

**Comment by brunobat** [comment]
If you want to disable tracing it's better just to disable the OTel SDK. You can now do it in runtime, per environment with the new [quarkus.otel.sdk.disabled](https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry_quarkus.otel.sdk.disabled) property.
Or like Roberto mentioned, at build time, but then it cannot be enabled without a new build.

**Reply by fabrii** [reply]
`quarkus.otel.exporter.otlp.enabled` seems a good fit. I still want the spans/baggage to be generated and propagated through API rest invocations.

Thanks!

**Comment by famod** [comment]
Sorry to revive this old discussion, but to me it feels like there is something missing:
- I have some environments where I cannot do an OTLP export (yet)
- still I want to have trace- and span-ids in my logfiles, so `quarkus.otel.sdk.disabled` is not for me
- I cannot use `quarkus.otel.exporter.otlp.enabled` because I build my app only once for all environments

So I'm missing a runtime property to disable the OTLP export. WDYT @radcortez @brunobat?

PS: Setting the `*.endpoint` properties to empty doesn't help.

**Reply by brunobat** [reply]
You can set `quarkus.otel.traces.exporter=none`
See: https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry_quarkus.otel.traces.exporter
All will be on, but no export will happen.

**Reply by famod** [reply]
Thanks, didn't know that one, but since it's also build time it doesn't work for me.

**Reply by brunobat** [reply]
You can try something else... Try to use the sampler to sample out everything:
build time prop: `quarkus.otel.traces.sampler=traceidratio`
runtime time prop: `quarkus.otel.traces.sampler.arg=1.0` -> On
runtime time prop: `quarkus.otel.traces.sampler.arg=0.0` -> Off

This one I've never tried it myself.

**Reply by famod** [reply]
@brunobat nice, this seems to work as I need it to. Thanks!

WDYT about a proper runtime property to disable OTLP export?

**Reply by famod** [reply]
Ah, and one more question: is it  `traceidratio` or `parentbased_traceidratio`, given that the default is `parentbased_always_on`?

**Reply by brunobat** [reply]
`parentbased*` will first take into account the parent span’s sampling decision.
From what I understand you want all or nothing regardless of the parent, therefore `traceidratio`should be better.

I think the sampler really addresses this requirement. 
Even when all your services are served by an OTel Collector, most likely you don't want to export all the spans, because of the high volume. Sampling with the right ratio will save storage without seriously harming analysis.

**Reply by famod** [reply]
Ok yeah, good point about the volume.

I'll re-check the docs, but maybe we should add a little hint that you can use that sampler threshold to switch off the export entirely at runtime.

**Reply by brunobat** [reply]
Definitely, it's a really good use case and it's not in the docs.

**Reply by brunobat** [reply]
Added this @famod: https://github.com/quarkusio/quarkus/pull/35789
Please let me know if you have suggestions.

**Reply by famod** [reply]
Looking good, thanks!
